StripeConnect/StripeConnectTests/Internal/Webview/ConnectWebViewTests.swift

```diff
@@ -0,0 +1,167 @@
+    var mockURLOpener: MockURLOpener!
+    var webView: ConnectWebView!
+    
+    override func setUp() {
+        mockURLOpener = .init()
+        webView = ConnectWebView(frame: .zero,
+                                 configuration: .init(),
+                                 urlOpener: mockURLOpener,
+                                 bundleVersion: "1.2.3")
+    }
+    
+    func testUserAgent() {
+        // Create an expectation for the asynchronous operation
+        let expectation = XCTestExpectation(description: "User Agent Fetched")
+        webView.evaluateJavaScript("navigator.userAgent") { (result, error) in
+            if let error = error {
+                XCTFail("Error fetching user agent: \(error.localizedDescription)")
+                return
+            }
+            
+            guard let userAgent = result as? String else {
+                XCTFail("User agent is not a string")
+                return
+            }
+                        
+            XCTAssertTrue(userAgent.hasSuffix("- stripe-ios/1.2.3"), "User agent should include the SDK identifier")
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 2.0)
+    }
+    
+    func testOpenSafariVCIfNotInAllowedHosts() {
+        var safariVC: SFSafariViewController?
+        webView.presentPopup = { vc in
+            safariVC = vc as? SFSafariViewController
+        }
+        let webView = webView.webView(webView,
+                        createWebViewWith: webView.configuration,
+                        for: MackNavigationAction(request: .init(url: URL(string: "https://google.com")!)),
+                        windowFeatures: .init())
+                
+        XCTAssertNil(webView)
+        XCTAssertNotNil(safariVC)
+    }
+    
+    func testOpenAsPopUpIfInAllowedHosts() {
+        for domain in [
+            "https://connect-js.stripe.com",
+            "https://connect-js.stripe.com/hello",
+            "https://connect.stripe.com/",
+            "https://connect.stripe.com/test",
+            "http://connect-js.stripe.com",
+            "http://connect-js.stripe.com/hello",
+            "http://connect.stripe.com/",
+            "http://connect.stripe.com/test"
+        ] {
+            var popUp: PopupWebViewController?
+            webView.presentPopup = { vc in
+                popUp = (vc as? UINavigationController)?.viewControllers.first as? PopupWebViewController
+                
+            }
+            let webView = webView.webView(webView,
+                                          createWebViewWith: webView.configuration,
+                                          for: MackNavigationAction(request: .init(url: URL(string: domain)!)),
+                                          windowFeatures: .init())
+            
+            XCTAssertEqual(popUp?.webView, webView)
+            XCTAssertNotNil(popUp?.navigationItem.rightBarButtonItem)
+            XCTAssertEqual(popUp?.webView.bundleVersion, "1.2.3")
```

Comment:
nit: if this fails, we're not going to know which URL it failed for in the array. Adding it to the failure message will help us debug faster:
```suggestion
            XCTAssertEqual(popUp?.webView, webView, domain)
            XCTAssertNotNil(popUp?.navigationItem.rightBarButtonItem, domain)
            XCTAssertEqual(popUp?.webView.bundleVersion, "1.2.3", domain)
```